poprox-db/migrations/versions/2024_09_04_1125-8bf414e0ddfb_remove_name_column_from_datasets_table.py

```diff
@@ -0,0 +1,29 @@
+"""remove name column from datasets table
+
+Revision ID: 8bf414e0ddfb
+Revises: dd50d8e7777e
+Create Date: 2024-09-04 11:25:43.062068
+
+"""
+
+from typing import Sequence, Union
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "8bf414e0ddfb"
+down_revision: Union[str, None] = "dd50d8e7777e"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.drop_column("datasets", "dataset_name")
```

Comment:
I don't have a problem with this change, but it does feel unconnected from the other changes, and I'm not immediately seeing the reasoning here.

Reply:
Each experiment is owned by a team and corresponds to a dataset, which (among other things) is how we know which account aliases to use when exporting experiment data. Loading a manifest is currently the only way to create a dataset, and the manifest doesn't have a field for naming the dataset so it's not clear what to put in this column.

An alternative change would be to make this column nullable if we expect other ways of creating datasets that would allow us to provide reasonable names.

Reply:
I think we're running into an overload of the term dataset. My gut says that outside experiment-team-tied datasets we may one-day have "public" datasets.

I don't think that hypothetical is worth keeping a column you're unsure about -- if we hit the hypothetical future where we have "public" datasets, let's solve that problem later.

Reply:
Public vs private is an interesting wrinkle too, but I was actually thinking of something a little different:

We've designed the database schema so that every experiment has a dataset, but not all datasets have experiments. That leaves us room to do things like export data for an experimenter so they can check properties of the data in order to find out if our platform makes sense for their experiment (like @sophiasun0515 has wanted to do re: domestic/international news bias.) Exporting data that way should also create a dataset with associated account id aliases but won't involve importing a manifest since it's pre-experiment.